provider/src/main/java/com/raygun/raygun4android/CrashReporting.java

```diff
@@ -0,0 +1,223 @@
+package com.raygun.raygun4android;
+
+import android.content.ComponentName;
+import android.content.Intent;
+import android.os.Build;
+
+import com.google.gson.Gson;
+import com.raygun.raygun4android.messages.crashreporting.RaygunBreadcrumbMessage;
+import com.raygun.raygun4android.messages.crashreporting.RaygunMessage;
+import com.raygun.raygun4android.network.RaygunNetworkUtils;
+import com.raygun.raygun4android.services.CrashReportingPostService;
+import com.raygun.raygun4android.utils.RaygunFileFilter;
+import com.raygun.raygun4android.utils.RaygunFileUtils;
+import com.raygun.raygun4android.utils.RaygunUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class CrashReporting {
+
+    private static RaygunUncaughtExceptionHandler handler;
+    private static CrashReportingOnBeforeSend onBeforeSend;
+    private static List tags;
+    private static Map userCustomData;
+    private static List<RaygunBreadcrumbMessage> breadcrumbs;
+    private static boolean shouldProcessBreadcrumbLocation = false;
+
+    static RaygunUncaughtExceptionHandler getExceptionHandler() {
+        return CrashReporting.handler;
+    }
+
+    static void setOnBeforeSend(CrashReportingOnBeforeSend onBeforeSend) {
+        CrashReporting.onBeforeSend = onBeforeSend;
+    }
+
+    static List getTags() {
+        return CrashReporting.tags;
+    }
+
+    static void setTags(List tags) {
+        CrashReporting.tags = tags;
+    }
+
+    static Map getUserCustomData() {
+        return CrashReporting.userCustomData;
+    }
+
+    static void setUserCustomData(Map userCustomData) {
```

Comment:
I feel that we should start removing "User" when referencing custom data. Every discussion we have at Raygun we never say UserCustomData, it's always CustomData. I think we should remove it for the 4.0.0 release. I am planning on redoing the API spec for providers this year and thats one thing I want to change. The json payload will still need send it as "userCustomData" but the API can have it removed.

Reply:
#57